src/app/screens/Home/index.tsx

```diff
@@ -62,12 +62,14 @@ const Home: FC = () => {
       if (currentUrl) {
         const url = new URL(currentUrl);
         setCurrentUrl(url);
-      }
 
-      if (currentUrl && currentUrl.startsWith("http")) {
-        browser.tabs.sendMessage(tabs[0].id as number, {
-          action: "extractLightningData",
-        });
+        if (currentUrl.startsWith("http")) {
+          browser.tabs.sendMessage(tabs[0].id as number, {
+            action: "extractLightningData",
+          });
+        }
+      } else {
+        setLoadingAllowance(false);
```

Comment:
Why do we need to set that explicitly? 

`loadAllowance` should only be triggered as the currentUrl is set? 

https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/Home/index.tsx#L46-L48

Reply:
But in cases where there's no `currentUrl` to be set i.e. Start Page in Safari, we go into a state where the `useEffect` is not invoked and the `loadAllowance` is never called, resulting in `loadingAllowance` not changing at all (initially set to true, which means it's loading but actually it isn't as the currentUrl is already set (to `null`)) and showing blank screen.

Reply:
I think that's somewhat strange as it's not at all obvious why this is being done. 

I'd probably try to get rid of the useEffect() and instead just call loadAllowance() in here. It's the only call, anyway and it makes things more obvious imho. (instead of having a hidden dependency make it explicit)